doc/crypto/BIO_new.pod

```diff
@@ -51,6 +51,12 @@ in a memory leak.
 Calling BIO_free_all() on a single BIO has the same effect as calling BIO_free()
 on it other than the discarded return value.
 
+=head1 HISTORY
+
+BIO_set() is deprecated since OpenSSL 1.1.0 as : 
+ - BIO type is now opaque, so no more stack-allocable.
+ - the new locking API now is now working per-object, and not per type.
```

Comment:
I don't think we normally add this kind of commentary in HISTORY...especially internal implementation details. I'd say its enough just to say its deprecated.